src/libmongoc/src/mongoc/mongoc-handshake.c

```diff
@@ -628,29 +623,37 @@ mongoc_handshake_data_append (const char *driver_name,
       return false;
    }
 
+   BSON_ASSERT (_mongoc_handshake_get ()->platform);
+
    /* allow practically any size for "platform", we'll trim it down in
     * _mongoc_handshake_build_doc_with_application */
    platform_space =
       HANDSHAKE_MAX_SIZE - (int) strlen (_mongoc_handshake_get ()->platform);
 
-   /* we check for an empty string as a special case to avoid an unnecessary
-    * delimiter being added in front of the string by _append_and_truncate */
-   if (strcmp (_mongoc_handshake_get ()->platform, "") == 0) {
-      bson_free (_mongoc_handshake_get ()->platform);
-      _mongoc_handshake_get ()->platform =
-         bson_strdup_printf ("%.*s", platform_space, platform);
-   } else {
-      _append_and_truncate (
-         &_mongoc_handshake_get ()->platform, platform, HANDSHAKE_MAX_SIZE);
+   if (platform) {
+      /* we check for an empty string as a special case to avoid an unnecessary
+       * delimiter being added in front of the string by _append_and_truncate */
+      if (strcmp (_mongoc_handshake_get ()->platform, "") == 0) {
```

Comment:
note: I'm ok with this, although I wonder if it might have been more immediately clear to see a direct check for the first byte being NULL, rather than strcmp().

Reply:
I considered this transformation as well. Since you pointed it out, I think I will include it in this PR (might as well).